partner_relations/models/res_partner_relation_all.py

```diff
@@ -0,0 +1,442 @@
+                )
+            return warning
+
+        type_selection_domain = []
+        if self.this_partner_id:
+            type_selection_domain += [
+                '|',
+                ('contact_type_this', '=', False),
+                ('contact_type_this', '=',
+                 self.this_partner_id.get_partner_type()),
+                '|',
+                ('partner_category_this', '=', False),
+                ('partner_category_this', 'in',
+                 self.this_partner_id.category_id.ids),
+            ]
+        if self.other_partner_id:
+            type_selection_domain += [
+                '|',
+                ('contact_type_other', '=', False),
+                ('contact_type_other', '=',
+                 self.other_partner_id.get_partner_type()),
+                '|',
+                ('partner_category_other', '=', False),
+                ('partner_category_other', 'in',
+                 self.other_partner_id.category_id.ids),
+            ]
+        result = {'domain': {
+            'type_selection_id': type_selection_domain,
+        }}
+        # Check wether domain results in no choice or wrong choice for
+        # type_selection_id:
+        warning = check_type_selection_domain(type_selection_domain)
+        if warning:
+            result['warning'] = warning
+        return result
+
+    @api.model
+    def _correct_vals(self, vals):
+        """Fill left and right partner from this and other partner."""
+        vals = vals.copy()
+        if 'this_partner_id' in vals:
+            vals['left_partner_id'] = vals['this_partner_id']
+            del vals['this_partner_id']
+        if 'other_partner_id' in vals:
+            vals['right_partner_id'] = vals['other_partner_id']
+            del vals['other_partner_id']
+        if 'type_selection_id' not in vals:
+            return vals
+        selection = self.type_selection_id.browse(vals['type_selection_id'])
+        type_id = selection.type_id.id
+        is_inverse = selection.is_inverse
+        vals['type_id'] = type_id
+        del vals['type_selection_id']
+        # Need to switch right and left partner if we are in reverse id:
+        if 'left_partner_id' in vals or 'right_partner_id' in vals:
+            if is_inverse:
+                left_partner_id = False
+                right_partner_id = False
+                if 'left_partner_id' in vals:
+                    right_partner_id = vals['left_partner_id']
+                    del vals['left_partner_id']
+                if 'right_partner_id' in vals:
+                    left_partner_id = vals['right_partner_id']
+                    del vals['right_partner_id']
+                if left_partner_id:
+                    vals['left_partner_id'] = left_partner_id
+                if right_partner_id:
+                    vals['right_partner_id'] = right_partner_id
+        return vals
+
+        def check_type_selection_domain(type_selection_domain):
```

Comment:
this should be quite different. Thanks for noticing! Something must have blown up during the rebasing, will investigate